org.thecrossingchurch.CustomJobs/Jobs/HubspotIntegrationMatching.cs

```diff
@@ -0,0 +1,718 @@
+            var contactResults = JsonConvert.DeserializeObject<HSContactQueryResult>(contactResponse.Content);
+            // Original - Contacts with emails that do not already have Rock IDs in the desired business unit
+            // Now - LOOK AT EVERY CONTACT
+            // contacts.AddRange(contactResults.results.Where(c => c.properties.email != null && c.properties.email != "" && ( c.properties.rock_person_id == null || c.properties.rock_person_id == "" || c.properties.rock_person_id == "0" )).ToList());
+            contacts.AddRange(contactResults.results);
+            if ( contactResults.paging != null && contactResults.paging.next != null && !String.IsNullOrEmpty(contactResults.paging.next.link) && request_count < 2000 )
+            {
+                GetContacts(contactResults.paging.next.link);
+            }
+        }
+
+        private List<int> FindPersonIds(HSContactResult contact, RockContext _context)
+        {
+            using ( RockContext context = new RockContext() )
+            {
+                SqlParameter[] sqlParams = new SqlParameter[] {
+                    new SqlParameter( "@first_name", contact.properties.firstname ?? "" ),
+                    new SqlParameter( "@last_name", contact.properties.lastname ?? "" ),
+                    new SqlParameter( "@email", contact.properties.email ?? "" ),
+                    new SqlParameter( "@mobile_number", contact.properties.phone ?? "" ),
+                };
+                var query = context.Database.SqlQuery<int>($@"
+                    DECLARE @matches int = (SELECT COUNT(*) FROM Person WHERE Email = @email);
+
+                    SELECT DISTINCT Person.Id
+                    FROM Person
+                            LEFT OUTER JOIN PhoneNumber ON Person.Id = PhoneNumber.PersonId
+                    WHERE ((@email IS NOT NULL AND @email != '') AND
+                        (Email = @email AND
+                            (((@first_name IS NULL OR @first_name = '') AND (@last_name IS NULL OR @last_name = '') AND @matches = 1) OR
+                            ((@first_name IS NOT NULL AND @first_name != '' AND
+                            (FirstName = @first_name OR NickName = @first_name)) OR
+                            (@last_name IS NOT NULL AND @last_name != '' AND LastName = @last_name) OR
+                            (@mobile_number IS NOT NULL AND @mobile_number != '' AND
+                            (Number = @mobile_number OR PhoneNumber.NumberFormatted = @mobile_number))))))
+                    ", sqlParams).ToList();
+                return query;
+            }
+        }
+
+        private int[] GetAllPersonIds(RockContext _context)
+        {
+            using ( RockContext context = new RockContext() )
+            {
+                var query = context.Database.SqlQuery<int>($@"
+                    SELECT Id
+                    FROM [dbo].[Person]
+                    Where Age >= 18
+                    AND RecordStatusValueId = 3
```

Comment:
Can we add criteria for only getting records with a `RecordTypeValueId` of 1 (Person) since we don't need businesses, nameless people, Rest keys, etc.

Can we also filter out people where `IsDeceased` is 1 (true)? The record status should be inactive for all of these people, but that isn't a guarantee.

Also, can we filter out people where `IsLockedAsChild` is 1 (true) since they are not classified as adults in our database?